lib/Dialect/TritonGPU/Transforms/WarpSpecialization/PartitionLoops.cpp

```diff
@@ -25,12 +25,15 @@ using Partition = WarpSchedule::Partition;
 // partition or the provided `partition`.
 static void eraseOtherPartitions(scf::ForOp &loop, const WarpSchedule &schedule,
                                  const Partition *partition) {
+  auto inPartition = [&](Operation *op) {
+    const Partition *opPartition = schedule.getPartition(op);
```

Comment:
```suggestion
    const Partition *opPartition = schedule.getPartition(loop.getBody()->findAncestorOpInBlock(*op));
```

Reply:
This should be sufficient to fix the problem. Nested ops should not have partitions assigned to them. Only ops immediately nested inside a loop. (For example, it doesn't make sense for an IfOp assigned to a partition to have nested ops assigned to different partitions).